### PR TITLE
Map armor changes

### PR DESCRIPTION
**Pull request by MattiDragon**
Not much to say. It's a registry now and entries are used all over the place (caused some args to not match). Shouldn't have much impact besides every mod having to rewrite their armor materials.

### CODE REVIEW

File: mappings/net/minecraft/item/ArmorItem.mapping [hunk 1]
```diff
@@ -18,8 +20,12 @@ CLASS net/minecraft/class_1738 net/minecraft/item/ArmorItem
...
 		FIELD field_41939 name Ljava/lang/String;
+		FIELD field_48839 CODEC Lcom/mojang/serialization/Codec;
 		METHOD <init> (Ljava/lang/String;ILnet/minecraft/class_1304;Ljava/lang/String;)V
 			ARG 3 equipmentSlot
 			ARG 4 name
 		METHOD method_48399 getEquipmentSlot ()Lnet/minecraft/class_1304;
 		METHOD method_48400 getName ()Ljava/lang/String;
+		METHOD method_56690 getMaxDamage (I)I
+			ARG 1 multiplier
+		METHOD method_56691 isTrimable ()Z
```
Comment (YanisBft) [inline mappings/net/minecraft/item/ArmorItem.mapping]:
```suggestion
		METHOD method_56691 isTrimmable ()Z
```

File: mappings/net/minecraft/item/ArmorMaterial.mapping [hunk 2]
```diff
@@ -55,3 +56,22 @@ CLASS net/minecraft/class_1741 net/minecraft/item/ArmorMaterial
 		COMMENT @return the amount of armor protection points offered by an {@link ArmorItem} with this {@link ArmorMaterial}
 		ARG 1 type
 			COMMENT the {@link ArmorItem.Type} of the {@link Item} with this {@link ArmorMaterial}
+	CLASS class_9196 Layer
+		FIELD field_48841 id Lnet/minecraft/class_2960;
+		FIELD field_48842 suffix Ljava/lang/String;
+		FIELD field_48843 dyable Z
```
Comment (Shnupbups) [inline mappings/net/minecraft/item/ArmorMaterial.mapping]:
```suggestion
		FIELD field_48843 dyeable Z
```

File: mappings/net/minecraft/item/ArmorMaterial.mapping [hunk 3]
```diff
@@ -55,3 +56,22 @@ CLASS net/minecraft/class_1741 net/minecraft/item/ArmorMaterial
...
+		FIELD field_48844 layer2Texture Lnet/minecraft/class_2960;
+		FIELD field_48845 layer1Texture Lnet/minecraft/class_2960;
+		METHOD <init> (Lnet/minecraft/class_2960;)V
+			ARG 1 id
+		METHOD <init> (Lnet/minecraft/class_2960;Ljava/lang/String;Z)V
+			ARG 1 id
+			ARG 2 suffix
+			ARG 3 dyable
+		METHOD method_56692 isDyable ()Z
```
Comment (Shnupbups) [inline mappings/net/minecraft/item/ArmorMaterial.mapping]:
```suggestion
		METHOD method_56692 isDyeable ()Z
```

**Review by Shnupbups — DISMISSED** [review]

File: mappings/net/minecraft/item/session
Comment (YanisBft) [inline mappings/net/minecraft/item/session:1]: to delete

**Review by YanisBft — APPROVED** [review]

**Review by apple502j — APPROVED** [review]

**Pull request merged**